components/dashboard/RequestView.tsx

```diff
@@ -0,0 +1,304 @@
+              setSelectedRequest(Request.Received);
+            }}
+          >
+            Received
+          </TabPrimary>
+          <TabPrimary
+            onClick={() => {
+              setSelectedRequest(Request.Sent);
+            }}
+          >
+            Sent
+          </TabPrimary>
+          <Spacer />
+          <Box marginTop="8px" marginBottom="16px">
+            <DashboardSearchInput />
+          </Box>
+        </TabList>
+        <TabPanels>
+          <TabPanel>
+            <RequestsTable
+              tenantRequests={tenantRequestList}
+              userId={user?.uid}
+              requestType="received"
+            />
+          </TabPanel>
+          <TabPanel>
+            <RequestsTable
+              tenantRequests={tenantRequestList}
+              userId={user?.uid}
+              requestType="sent"
+            />
+          </TabPanel>
+        </TabPanels>
+      </Tabs>
+    </DashboardCard>
+  );
+}
+
+const DeleteConfirmationModal = (props: DeleteProps) => {
+  const { isOpen, onClose, request } = props;
+
+  const listingTitle = request?.listing.data.details.title;
+
+  const toast = useToast();
+
+  const handleDelete = async (requestId: string | undefined) => {
+    try {
+      await tenantRequests.doc(requestId).delete();
+      toast({
+        title: "Deleted Tenant Request",
+        description: `We successfully deleted your tenant Request for listing: ${listingTitle}`,
+        isClosable: true,
+        duration: 4000,
+        status: "success",
+      });
+    } catch (e) {
+      toast({
+        title: "Something went wrong",
+        description:
+          "An error occurred and we couldn't delete your tenant request. Please try again later.",
+        isClosable: true,
+        duration: 4000,
+        status: "error",
+      });
+    }
+  };
+
+  const onDelete = (requestId: string | undefined) => {
+    handleDelete(requestId);
+    onClose();
+  };
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} isCentered size="sm">
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>
+          <Heading5>Delete Listing</Heading5>
+        </ModalHeader>
+        <ModalCloseButton />
+        <ModalBody mb="1rem">
+          <p>Are you sure you want to delete the tenant request for listing,</p>
+          <b>{listingTitle}</b>?<p>This action cannot be undone.</p>
+        </ModalBody>
+        <ModalFooter>
+          <ButtonPrimary mr={3} onClick={() => onDelete(request?.id)}>
+            Delete
+          </ButtonPrimary>
+          <ButtonSecondary onClick={onClose}>Cancel</ButtonSecondary>
+        </ModalFooter>
+      </ModalContent>
+    </Modal>
+  );
+};
+
+const RequestsTable = (props: {
+  tenantRequests?: TenantRequestEntry[];
+  userId: string | undefined;
+  requestType: string;
+}) => {
+  const { tenantRequests, userId, requestType } = props;
+
+  const [tenantRequest, setTenantRequest] = useState<
+    TenantRequestEntry | undefined
```

Comment:
Nitpick: I'm not sure but the `| undefined` might be unnecessary here?

Reply:
Throws and error when I try to remove it, I think there is a situation where it can be undefined

Reply:
Oops, ignore my comment then :laughing: